### PR TITLE
[ARCTIC-1478][AMS] Adapting IcebergCatalogImpl: GlueCatalog

### LINKED ISSUE #1478: [Subtask]: Iceberg GlueCatalog Adaptation
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### Description

_No response_

### Parent issue

- #1477

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by XBaith**
## Why are the changes needed?

fix #1478 

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Review by baiyangtx — APPROVED** [review]
LGTM

**Comment by baiyangtx** [comment]
I tend to wait until the refactor-ams branch is merged into master, then changing this PR directly to the master branch.

File: core/pom.xml
Comment (shidayang) [inline core/pom.xml:204]: It would be better to add these dependencies to parent <dependencyManagement>

Comment (baiyangtx) [reply core/pom.xml:204]: If moving to parent, it will be depended by all modules.

**Pull request merged**
